src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java

```diff
@@ -553,24 +552,27 @@ private void fillmodelColumnHandles(List<ColumnFamilyHandle> familyHandles) thro
     private interface MyRunnable<R> {
         R run() throws Exception;
     }
+
     private void fillMissingColumns(List<ColumnFamilyDescriptor> familyDescriptors, List<ColumnFamilyHandle> familyHandles, String path) throws Exception {
-        List<ColumnFamilyDescriptor> columnFamilies = RocksDB.listColumnFamilies(new Options().setCreateIfMissing(true), path)
-                .stream()
-                .map(b -> new ColumnFamilyDescriptor(b, new ColumnFamilyOptions()))
-                .collect(Collectors.toList());
-        columnFamilies.add(0, familyDescriptors.get(0));
-        List<ColumnFamilyDescriptor> missingFromDatabase = familyDescriptors.stream().filter(d -> columnFamilies.stream().filter(desc -> new String(desc.columnFamilyName()).equals(new String(d.columnFamilyName()))).toArray().length == 0).collect(Collectors.toList());
-        List<ColumnFamilyDescriptor> missingFromDescription = columnFamilies.stream().filter(d -> familyDescriptors.stream().filter(desc -> new String(desc.columnFamilyName()).equals(new String(d.columnFamilyName()))).toArray().length == 0).collect(Collectors.toList());
-        if (missingFromDatabase.size() != 0) {
-            missingFromDatabase.remove(familyDescriptors.get(0));
-            db = RocksDB.open(options, path, columnFamilies, familyHandles);
-            for (ColumnFamilyDescriptor description : missingFromDatabase) {
-                addColumnFamily(description.columnFamilyName(), db);
+        try (final Options options = new Options().setCreateIfMissing(true)) {
```

Comment:
not sure what `Options` are, but it doesn't sound like this statement will fail at all
`new Options().setCreateIfMissing(true)`

are `Option` a closeable resource?

Reply:
I see, it's an Option for RocksDB, still having a hard time looking at the class family tree and finding an implementation or extension of an option being a resource.
https://github.com/facebook/rocksdb/blob/master/java/src/main/java/org/rocksdb/Options.java


Why the `try(final Options options ...)`

Reply:
The inheritance of Options shows that it's a reference to an underlying C++ object and an AutoClosable. So it's pointing on a resource somewhere else and should be closed in any case.